### PR TITLE
Added support for DNS and DNS Search

### PR DESCRIPTION
**Pull request by FrenchBen**
Add support for custom DNS and Search domains within the advance tab.

### CODE REVIEW

**Comment by jeffdm** [comment]
Code LGTM

@mchiang0610 can you take a look?

**Comment by mchiang0610** [comment]
If I run the demo nginx container in Kitematic and change the DNS to 8.8.8.8 (Google's DNS), the preview screen goes empty (correct behavior since it should not contact outside) but when I click open browser icon (or anywhere within), I get this error: 

>> [9972:0828/163650:INFO:CONSOLE(59)] "Uncaught TypeError: Error processing argument 0.", source: /Users/mc/github/kitematic/build/components/ContainerHomePreview.react.js (59)

@FrenchBen

**Comment by FrenchBen** [comment]
@mchiang0610 I'll check to see what's going on with it - It shouldn't affect the preview screen as it's only modifying what's in the resolv.conf - I'll do some debugging. 

Thanks for the QA.

**Comment by FrenchBen** [comment]
Cleaned up a lot of the Docker util start options - By default *no* option is allowed on 'start' but the dockerode module does allow the over-write on start of the container's settings. 

I left Bind in there, as I believe it was there originally, but it should really be moved to the createcontainer method.

File: src/components/ContainerSettingsAdvanced.react.js [hunk 1]
```diff
@@ -1,8 +1,12 @@
+import _ from 'underscore';
 import React from 'react/addons';
 import metrics from '../utils/MetricsUtil';
 import ContainerUtil from '../utils/ContainerUtil';
 import containerActions from '../actions/ContainerActions';
 
+var ipRegex = /\b(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\b/;
+var domainRegex = /([a-z0-9-]+\.(?:com|net|org|co\.uk|localhost|dev))(?:\/|$)/;
```
Comment (fd) [inline src/components/ContainerSettingsAdvanced.react.js]: I think this is overly restrictive. The web is not limited to just `.com`,  `.net`, `.org`, `.co.uk`, `.localhost` and `.dev`.
Maybe something like `/^([a-z0-9-]+(?:\.[a-z0-9-]+)+)(?:\/|$)/` is be better.

Comment (FrenchBen) [inline src/components/ContainerSettingsAdvanced.react.js]: Good suggestion, although a lot looser, but should work nicely - I'll make the proper updates and will push it.

**Comment by FrenchBen** [comment]
#closing due to lack of interest in DNS settings

**Pull request closed without merging**